common/lib/common-utils/src/typedEventEmitter.ts

```diff
@@ -36,6 +39,58 @@ export type TypedEventTransform<TThis, TEvent> =
         // Add the default overload so this is covertable to EventEmitter regardless of environment
         TransformedEvent<TThis, EventEmitterEventType, any[]>;
 
+/** Converting from IEvent shape to EventSpec shape. Might allow for easier transition to EventSpec */
+export type ToEventArgsMapping<TEvent> =
+    //* TODO: ...Start with all 15 like IEventTransformer
+    TEvent extends {
+        (event: infer E0, listener: (...args: infer A0) => void);
+        (event: infer E1, listener: (...args: infer A1) => void);
+        (event: infer E2, listener: (...args: infer A2) => void);
+        (event: infer E3, listener: (...args: infer A3) => void);
+        (event: string, listener: (...args: any[]) => void);
+    }
+        ? SingleEventArgsMapping<E0, A0> &
+              SingleEventArgsMapping<E1, A1> &
+              SingleEventArgsMapping<E2, A2> &
+              SingleEventArgsMapping<E3, A3>
+        : TEvent extends {
+              (event: infer E0, listener: (...args: infer A0) => void);
+              (event: infer E1, listener: (...args: infer A1) => void);
+              (event: infer E2, listener: (...args: infer A2) => void);
+              (event: string, listener: (...args: any[]) => void);
+          }
+        ? SingleEventArgsMapping<E0, A0> &
+              SingleEventArgsMapping<E1, A1> &
+              SingleEventArgsMapping<E2, A2>
+        : TEvent extends {
+              (event: infer E0, listener: (...args: infer A0) => void);
+              (event: infer E1, listener: (...args: infer A1) => void);
+              (event: string, listener: (...args: any[]) => void);
+          }
+        ? SingleEventArgsMapping<E0, A0> & SingleEventArgsMapping<E1, A1>
+        : TEvent extends {
+              (event: infer E0, listener: (...args: infer A0) => void);
+              (event: string, listener: (...args: any[]) => void);
+          }
+        ? SingleEventArgsMapping<E0, A0>
+        : SingleEventArgsMapping<string, any[]>;
+
+type SingleEventArgsMapping<TEventKey, TListenerArgs extends any[]> = TEventKey extends string
+    ? {
+          [TK in TEventKey]: TListenerArgs;
+      }
+    : never;
+
+type RequireAllStringKeys<TEventSpec> = keyof TEventSpec extends string ? keyof TEventSpec : never;
```

Comment:
Consider filtering out non-string keys rather than banning types with non-string keys.

Reply:
I'm going to pass on this, because I'm past my time-box for this work at this point, but would be happy to see a follow-up PR (or feel free to pull this branch!) if you want to try.  I know you had a "filter" approach in your PR, maybe that would work here too?

Reply:
Yea I'm happy to do it as a followup PR 😀 I think it'll be best to break up these PRs into the smallest chunks possible, since it's easy to fiddle with the types and accidentally break some edge case without realizing it.